Add com.snowplowanalytics.snowplow/gdpr/jsonschema/1-0-0

colmsnowplow opened the pull request.

knservis commented:
@colmsnowplow Can we call this gdpr_context instead of just gdpr for uniformity

colmsnowplow commented:
> @colmsnowplow Can we call this gdpr_context instead of just gdpr for uniformity

The standard is to leave it out. I can see there are some that have `_context` but those are the exception.

yalisassoon commented on schemas/com.snowplowanalytics.snowplow/gdpr/jsonschema/1-0-0:
```diff
@@ -0,0 +1,35 @@
+        },
+        "documentVersion": {
+          "type": ["string", "null"],
+          "maxLength": 5,
```
Suggest making this longer e.g. 16 characters, so you can accommodate e.g. `12.4.2`

colmsnowplow replied:
Yup should've thought of that, fixing.

colmsnowplow commented:
> LGTM, except from the name gdpr could mean anything gdpr_context is much more informative. I woudl definitely change the name.

I also prefer appending `_context` to the names of contexts. But the already established convention is to leave it out, so it doesn't make sense to break it. Most of the contexts follow this convention, and indeed it's what we recommend our customers follow in onboarding.

@chuwy I've never released to Iglu Central before, is there some process beyond asking you/someone else to merge the PR?

alexanderdean commented:
Looks fine to me. Does it not need JSON Paths and a Redshift table defn?

chuwy commented on CHANGELOG:
```diff
@@ -1,3 +1,8 @@
+Release 105 (2019-06-26)
+-----------------------
+
```
Missing newline.

colmsnowplow replied:
Fixed.

chuwy commented on schemas/com.snowplowanalytics.snowplow/gdpr/jsonschema/1-0-0:
```diff
@@ -0,0 +1,35 @@
+    "type": "object",
+    "properties": {
+        "basisForProcessing": {
+          "enum": ["consent", "contract", "legal obligation", "vital interest", "public task", "legitimate interests"],
```
Do we have a control over these values? I have a slight preference towards enums without spaces.

colmsnowplow commented:
Added JSONpath files, removed whitespace from enums and removed newline from CHANGELOG. 

Hoping this is GTG now but let me know if not @alexanderdean and @chuwy :)

alexanderdean commented:
Let's take the non-essential reviewers off this. If everybody else ticks, I can release tonight 👍

colmsnowplow commented:
Thanks Alex, all necessary approvals given now!

alexanderdean commented on schemas/com.snowplowanalytics.snowplow/gdpr/jsonschema/1-0-0:
```diff
@@ -0,0 +1,35 @@
+    "type": "object",
+    "properties": {
+        "basisForProcessing": {
+          "enum": ["consent", "contract", "legal_obligation", "vital_interest", "public_task", "legitimate_interests"],
```
Should be `vital_interests` note the trailing s for consistency with legitimate_interests. Confirmed on https://ico.org.uk/for-organisations/guide-to-data-protection/guide-to-the-general-data-protection-regulation-gdpr/lawful-basis-for-processing/

colmsnowplow replied:
Thanks for flagging @alexanderdean , fixed now. :)
